packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts

```diff
@@ -0,0 +1,366 @@
+import { IDropdownStyles, IDropdownStyleProps } from './Dropdown.types';
+import { IStyleFunction } from '../../Utilities';
+import {
+  FontSizes,
+  FontWeights,
+  HighContrastSelector,
+  IRawStyle,
+  IStyle,
+  getGlobalClassNames,
+  normalize
+} from '../../Styling';
+
+const GlobalClassNames = {
+  root: 'ms-Dropdown-container',
+  label: 'ms-Dropdown-label',
+  dropdown: 'ms-Dropdown',
+  title: 'ms-Dropdown-title',
+  caretDownWrapper: 'ms-Dropdown-caretDownWrapper',
+  caretDown: 'ms-Dropdown-caretDown',
+  callout: 'ms-Dropdown-callout',
+  panel: 'ms-Dropdown-panel',
+  dropdownItems: 'ms-Dropdown-items',
+  dropdownDivider: 'ms-Dropdown-divider',
+  dropdownOptionText: 'ms-Dropdown-optionText',
+  dropdownItemHeader: 'ms-Dropdown-header',
+  titleIsPlaceHolder: 'ms-Dropdown-titleIsPlaceHolder',
+  titleHasError: 'ms-Dropdown-title--hasError'
+};
+
+const DROPDOWN_HEIGHT = 32;
+const DROPDOWN_ITEMHEIGHT = 32;
+
+const highContrastAdjustMixin = {
+  // highContrastAdjust mixin
+  '@media screen and (-ms-high-contrast: active),  screen and (-ms-high-contrast: black-on-white)': {
+    MsHighContrastAdjust: 'none'
+  }
+};
+
+const highContrastItemAndTitleStateMixin: IRawStyle = {
+  selectors: {
+    [HighContrastSelector]: {
+      backgroundColor: 'Highlight',
+      borderColor: 'Highlight',
+      color: 'Highlight'
+    },
+    ...highContrastAdjustMixin
```

Comment:
There's no real advantage and in fact it is very likely worse since you defer the flattening of the array into the runtime call of getClassNames which occurs during every single render pass...